src/main/java/gov/cabinetoffice/gapfindapiadmin/services/ApiGatewayService.java

```diff
@@ -70,4 +70,14 @@ protected void saveKeyInDatabase(String keyName, CreateApiKeyResponse response,
         apiKeyService.saveApiKey(apiKey);
     }
 
+    public void deleteApiKey(String keyName) {
+        apiGatewayClient.getApiKeys().items()
```

Comment:
is there not a method in the apiGatewayClient that allows us to just get one key? This feels like it could get out of hand pretty quickly. Something like: apiGatewayClient.getApiKey(GetApiKeyRequest.builder().apiKey(apiKey.id()).build()). If that apikey in the end point isnt the same as the one we have, it might be useful to save an encrypted version of the api key in the db so we dont have to look through every single api key we have

Reply:
We don't store the apiKeyId from api gateway or the apiKey object so we can only find the key in api gateway through the key name